core/src/test/java/org/apache/calcite/test/SqlJsonFunctionsTest.java

```diff
@@ -0,0 +1,752 @@
+        SqlFunctions.PathContext.withStrictException(new Exception("test message")),
+        SqlJsonExistsErrorBehavior.TRUE,
+        is(true));
+
+    assertjsonExists(
+        SqlFunctions.PathContext.withStrictException(new Exception("test message")),
+        SqlJsonExistsErrorBehavior.UNKNOWN,
+        nullValue());
+
+    try {
+      SqlFunctions.
+          jsonExists(SqlFunctions.PathContext
+                  .withStrictException(new Exception("test message")),
+              SqlJsonExistsErrorBehavior.ERROR);
+      fail("expect exception, but not");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage(), is("java.lang.Exception: test message"));
+    }
+  }
+
+  @Test
+  public void testJsonValueAny() {
+    assertJsonValueAny(
+        SqlFunctions.PathContext
+            .withReturned(SqlFunctions.PathMode.LAX, "bar"),
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        is("bar")
+    );
+    assertJsonValueAny(
+        SqlFunctions.PathContext
+            .withReturned(SqlFunctions.PathMode.LAX, null),
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        nullValue()
+    );
+    assertJsonValueAny(
+        SqlFunctions.PathContext
+            .withReturned(SqlFunctions.PathMode.LAX, null),
+        SqlJsonValueEmptyOrErrorBehavior.DEFAULT,
+        "empty",
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        is("empty")
+    );
+    try {
+      SqlFunctions
+          .jsonValueAny(
+              SqlFunctions.PathContext
+                  .withReturned(SqlFunctions.PathMode.LAX, null),
+              SqlJsonValueEmptyOrErrorBehavior.ERROR,
+              null,
+              SqlJsonValueEmptyOrErrorBehavior.NULL,
+              null);
+      fail("expect exception, but not");
+    } catch (Exception e) {
+      assertThat(e.getMessage(), is("empty json value"));
+    }
+    assertJsonValueAny(
+        SqlFunctions.PathContext
+            .withReturned(SqlFunctions.PathMode.LAX, Collections.emptyList()),
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        nullValue()
+    );
+    assertJsonValueAny(
+        SqlFunctions.PathContext
+            .withReturned(SqlFunctions.PathMode.LAX, Collections.emptyList()),
+        SqlJsonValueEmptyOrErrorBehavior.DEFAULT,
+        "empty",
+        SqlJsonValueEmptyOrErrorBehavior.NULL,
+        null,
+        is("empty")
+    );
+    try {
+      SqlFunctions
+          .jsonValueAny(
+              SqlFunctions.PathContext
+                  .withReturned(SqlFunctions.PathMode.LAX, Collections.emptyList()),
+              SqlJsonValueEmptyOrErrorBehavior.ERROR,
+              null,
+              SqlJsonValueEmptyOrErrorBehavior.NULL,
+              null);
+      fail("expect exception, but not");
```

Comment:
It is sad that exception message does not clarify the reason.
There are "obvious" `fail(...)`, however this one is not clear from the first sight.

Reply:
@vlsi I have made individual failure check methods in latest commit, please check it.